### DISCUSSION TITLE
RFC: Add support for MAX32 microcontroller

### DISCUSSION THREAD

**Discussion by sampaio23** [micropython/micropython#16277] [Ideas]
Hello, I hope you are all doing well!

I've been working with the Analog Devices' MAX32 family of microcontrollers, specifically the MAX32655, and decided to experiment with MicroPython. I've implemented the basic REPL with the `mp_hal_stdin_rx_chr` and `mp_hal_stdout_tx_strn` and an example module using the _embed_ port. The ease of use and flexibility was amazing, very cool project!
I also presented a paper using MicroPython at [IECON 2024](https://www.iecon-2024.org/), although the paper is not yet published.

I would like to know if I can submit a PR with the basic port to the MAX32655 and continuously working on the other peripherals. I am willing to become the maintainer of the ports to the MAX32 family.

Thank you for your attention!

**Comment by bidrohini** [comment]
Include the core REPL implementation and any example modules you've already developed for the MAX32655. Ensure they compile and run reliably.

**Reply by sampaio23** [reply]
Thank you!

**Comment by MaureenHelm** [comment]
You can use the Zephyr port of MicroPython on MAX32 parts, including MAX32655. Full list of Zephyr-supported ADI boards here:
https://docs.zephyrproject.org/latest/boards/index.html#vendor=adi

**Reply by sampaio23** [reply]
Thank you, I didn't know there was a Zephyr port of MicroPython. I will try it!
